ui/dashboards/src/components/EditJsonDialog/EditJsonDialog.tsx

```diff
@@ -74,11 +77,14 @@ const EditJsonDialogForm = (props: EditJsonDialogProps) => {
             maxHeight="70vh"
             value={draftDashboard}
             onChange={(value: string) => completeDraftDashboard(value)}
+            readOnly={isReadonly}
           />
         </FormControl>
       </Dialog.Content>
       <Dialog.Actions>
-        <Dialog.PrimaryButton onClick={handleApply}>Apply</Dialog.PrimaryButton>
+        <Dialog.PrimaryButton disabled={isReadonly} onClick={handleApply}>
```

Comment:
I think, we could hide/remove it completely in readonly mode

Reply:
👍 I was hesitating to do so as in other places we went for disabling buttons instead of hiding them (e.g server in readonly mode, or not the right permissions), but in the end I think the use cases are different here so let's hide this one :)